postprocessing/janus-pp-rec.c

```diff
@@ -1171,15 +1187,28 @@ int main(int argc, char *argv[])
 			JANUS_LOG(LOG_VERB, "[%10lu][%4d] time=%"SCNu64"s\n", tmp->offset, tmp->len, tmp->ts);
 		tmp = tmp->next;
 	}
-	JANUS_LOG(LOG_INFO, "Counted %"SCNu32" frame packets\n", count);
-	if(rotated != -1) {
-		if(rotated == 0 && last_rotation != 0) {
-			JANUS_LOG(LOG_INFO, "The video is rotated\n");
-		} else if(rotated > 0) {
-			JANUS_LOG(LOG_INFO, "The video changed orientation %d times\n", rotated);
+	if(!ext_only) {
+		JANUS_LOG(LOG_INFO, "Counted %"SCNu32" frame packets\n", count);
+		if (rotated != -1) {
+			if (rotated == 0 && last_rotation != 0) {
+				JANUS_LOG(LOG_INFO, "The video is rotated\n");
+			} else if (rotated > 0) {
+				JANUS_LOG(LOG_INFO, "The video changed orientation %d times\n", rotated);
+			}
 		}
 	}
 
+	if (ext_only) {
+		/* Look for rotations changes */
+		janus_pp_detect_rotation_changes(report, list);
```

Comment:
I don't understand this. You're looping on the whole list again just to populate the report. Why not populate it _while_ we're parsing it, since we know the extensions? Is it to account for potentially out of order packets?